pil-analyzer/src/pil_analyzer.rs

```diff
@@ -287,8 +288,18 @@ impl PILAnalyzer {
                     if let Some(selector) = &mut part.selector {
                         expressions.push((selector, Type::Expr.into()))
                     }
-                    for e in &mut part.expressions {
-                        expressions.push((e, Type::Expr.into()))
+                    if let Expression::ArrayLiteral(_, _) = part.expressions.as_ref() {
```

Comment:
Why only for array literals? This should work for any expression. We can now also test things like `x $ f() in g()`.

Reply:
I seem to remember that we had talked about this kind of expressions now always being of the form x $ [f()] in [g()], but no problem, I can fix that. The only thing I need to know (if I understand correctly), is how to treat those expressions that are not arrays here:

https://github.com/powdr-labs/powdr/blob/7e3a5c70913537f6351064762eb82ff24abd0199/pil-analyzer/src/condenser.rs#L293

because this function expects &sel_expr.expressions to evaluate to an array (which doesn't happen anymore in the example you gave me).

Reply:
Sorry, I should have expanded the example a little: My idea was that `f()` and `g()` evaluate to arrays. So you don't have to support tuples or anything else in `condense_to_array_of_algebraic_expressions`.

What I meant was that you should not expect here in line 291 that part.expressions is syntactically an array literal - only that its type is an array of expr.